packages/core/src/content/content-client.ts

```diff
@@ -129,4 +137,183 @@ export class ContentClient {
     const response = await this.get<LocalesQueryResponse>(GET_LOCALES_QUERY);
     return response.manyLocale;
   }
+
+  /**
+   * Retrieves all available taxonomies.
+   *
+   * This method supports optional pagination for both taxonomies and their terms.
+   * - If `taxonomyPageSize` is provided, it limits the number of taxonomies returned per request.
+   * - If `termsPageSize` is provided, it limits the number of terms returned per taxonomy per request.
+   * - If either is omitted, the API defaults are used (returns all results in a single request if no pagination is required).
+   *
+   * @param {number} [taxonomyPageSize] - Optional. The number of taxonomies per page.
+   * @param {number} [termsPageSize] - Optional. The number of terms per page within each taxonomy.
+   * @returns A promise that resolves to a paginated list of taxonomies, including pagination helpers (`fetchNext`) if applicable.
+   */
+  async getTaxonomies(
+    taxonomyPageSize?: number,
+    termsPageSize?: number
+  ): Promise<ContentItemList<Taxonomy>> {
+    const fetchTaxonomiesPage = async (after: string = ''): Promise<ContentItemList<Taxonomy>> => {
+      debug.content(
+        'Fetching taxonomies (pageSize: %s, after: %s)',
+        taxonomyPageSize ?? 'API Default',
+        after
+      );
+
+      const variables: Record<string, any> = {
+        after: after || '',
```

Comment:
The fallback is defined on parameter level above
```suggestion
        after,
```